sdk/search/search-documents/src/generated/service/models/index.ts

```diff
@@ -1,4550 +1,3853 @@
+  | DataChangeDetectionPolicy
+  | HighWaterMarkChangeDetectionPolicy
+  | SqlIntegratedChangeTrackingPolicy;
+export type DataDeletionDetectionPolicyUnion =
+  | DataDeletionDetectionPolicy
+  | SoftDeleteColumnDeletionDetectionPolicy;
+export type SearchIndexerSkillUnion =
+  | SearchIndexerSkill
+  | ConditionalSkill
+  | KeyPhraseExtractionSkill
+  | OcrSkill
+  | ImageAnalysisSkill
+  | LanguageDetectionSkill
+  | ShaperSkill
+  | MergeSkill
+  | EntityRecognitionSkill
+  | SentimentSkill
+  | SplitSkill
+  | CustomEntityLookupSkill
+  | TextTranslationSkill
+  | WebApiSkill;
+export type CognitiveServicesAccountUnion =
+  | CognitiveServicesAccount
+  | DefaultCognitiveServicesAccount
+  | CognitiveServicesAccountKey;
+export type ScoringFunctionUnion =
+  | ScoringFunction
+  | DistanceScoringFunction
+  | FreshnessScoringFunction
+  | MagnitudeScoringFunction
+  | TagScoringFunction;
+export type LexicalAnalyzerUnion =
+  | LexicalAnalyzer
+  | CustomAnalyzer
+  | PatternAnalyzer
+  | LuceneStandardAnalyzer
+  | StopAnalyzer;
+export type LexicalTokenizerUnion =
+  | LexicalTokenizer
+  | ClassicTokenizer
+  | EdgeNGramTokenizer
+  | KeywordTokenizer
+  | KeywordTokenizerV2
+  | MicrosoftLanguageTokenizer
+  | MicrosoftLanguageStemmingTokenizer
+  | NGramTokenizer
+  | PathHierarchyTokenizerV2
+  | PatternTokenizer
+  | LuceneStandardTokenizer
+  | LuceneStandardTokenizerV2
+  | UaxUrlEmailTokenizer;
+export type TokenFilterUnion =
+  | TokenFilter
+  | AsciiFoldingTokenFilter
+  | CjkBigramTokenFilter
+  | CommonGramTokenFilter
+  | DictionaryDecompounderTokenFilter
+  | EdgeNGramTokenFilter
+  | EdgeNGramTokenFilterV2
+  | ElisionTokenFilter
+  | KeepTokenFilter
+  | KeywordMarkerTokenFilter
+  | LengthTokenFilter
+  | LimitTokenFilter
+  | NGramTokenFilter
+  | NGramTokenFilterV2
+  | PatternCaptureTokenFilter
+  | PatternReplaceTokenFilter
+  | PhoneticTokenFilter
+  | ShingleTokenFilter
+  | SnowballTokenFilter
+  | StemmerTokenFilter
+  | StemmerOverrideTokenFilter
+  | StopwordsTokenFilter
+  | SynonymTokenFilter
+  | TruncateTokenFilter
+  | UniqueTokenFilter
+  | WordDelimiterTokenFilter;
+export type CharFilterUnion =
+  | CharFilter
+  | MappingCharFilter
+  | PatternReplaceCharFilter;
+export type SimilarityUnion = Similarity | ClassicSimilarity | BM25Similarity;
+
+/** Represents a datasource definition, which can be used to configure an indexer. */
+export interface SearchIndexerDataSource {
+  /** The name of the datasource. */
+  name: string;
+  /** The description of the datasource. */
+  description?: string;
+  /** The type of the datasource. */
+  type: SearchIndexerDataSourceType;
+  /** Credentials for the datasource. */
+  credentials: DataSourceCredentials;
+  /** The data container for the datasource. */
+  container: SearchIndexerDataContainer;
+  /** The data change detection policy for the datasource. */
+  dataChangeDetectionPolicy?: DataChangeDetectionPolicyUnion | null;
```

Comment:
@xirzec One of the questions you had was - What is the reason for this ```| null``` value at the end of ```dataChangeDetectionPolicy```. To explain this, we need to go the swagger:

```json
....................
....................
"SearchIndexerDataSource": {
  "properties": {
    "name": {
      "externalDocs": {
        "url": "https://docs.microsoft.com/rest/api/searchservice/Naming-rules"
      },
      "type": "string",
      "description": "The name of the datasource."
    },
    "description": {
      "type": "string",
      "description": "The description of the datasource."
    },
    "type": {
      "$ref": "#/definitions/SearchIndexerDataSourceType",
      "description": "The type of the datasource."
    },
    "credentials": {
      "$ref": "#/definitions/DataSourceCredentials",
      "description": "Credentials for the datasource."
    },
    "container": {
      "$ref": "#/definitions/SearchIndexerDataContainer",
      "description": "The data container for the datasource."
    },
    "dataChangeDetectionPolicy": {
      "$ref": "#/definitions/DataChangeDetectionPolicy",
      "x-nullable": true,
      "description": "The data change detection policy for the datasource."
    },
    "dataDeletionDetectionPolicy": {
      "$ref": "#/definitions/DataDeletionDetectionPolicy",
      "x-nullable": true,
      "description": "The data deletion detection policy for the datasource."
    },
    "@odata.etag": {
      "x-ms-client-name": "ETag",
      "type": "string",
      "description": "The ETag of the data source."
    },
    "encryptionKey": {
      "$ref": "#/definitions/SearchResourceEncryptionKey",
      "description": "................",
      "externalDocs": {
        "url": "https://aka.ms/azure-search-encryption-with-cmk"
      },
      "x-nullable": true
    }
  },
  "required": [
    "name",
    "type",
    "credentials",
    "container"
  ],
  "description": "Represents a datasource definition, which can be used to configure an indexer."
}
```

As you can see from the above swagger, the dataChangeDetectionPolicy has the property ```x-nullable: true```. So, we need to add the ```| null``` value at the end. 

**What would happen if we do not add this?**
If the SDK user wants to actually send in the ```null``` value (as the service allows it - assuming the value ```null``` has some special property on the server side.), it might not be possible to send it. Also, to reflect the swagger code closely, we need to add this ```| null``` at the end.

Reply:
I'm fine supporting something the server does -- I'm just curious how the server handles it. If you set null on an update, is that the proper way to remove the thing? And how is that functionally different from us removing that part of the API shape?

What's confusing me is null is how you'd blank out part of the payload in C#, but in JS we have `undefined` to do this same job. It's rare that `undefined` and `null` have different meanings. If they mean the same thing, do we need both?

Reply:
To me, `nullable` should mean optional. We really don't want users to have to think about the differences between `{x: undefined}`, `{}`, and `{x: null}` do we? I hope that no service treats these differently.